src/logging.cpp

```diff
@@ -247,6 +249,25 @@ void BCLog::Logger::LogPrintStr(const std::string& str)
 
     str_prefixed = LogTimestampStr(str_prefixed);
 
+    // Rate limit logging to disk to avoid disk filling attacks.
+    bool skip_writing_to_disk_due_to_rate_limiting{false};
+    if (!skip_disk_usage_rate_limiting) {
+        const std::chrono::seconds now = GetTime<std::chrono::seconds>();
+        if ((now - m_last_reset_of_bytes_logged_per_source_location) > std::chrono::hours{1}) {
+            m_bytes_logged_per_source_location.clear();
+            m_last_reset_of_bytes_logged_per_source_location = now;
```

Comment:
It'd be good to check here whether logging was suppressed for this log, and if so log that logging is restarting for this log (eg. "Restarting logging from %s ...")

Reply:
That's a good idea. I'll implement!